StoreForecasts/Program.cs

```diff
@@ -1,2 +1,93 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿
+using System;
+using System.Collections.Generic;
+using dotenv.net;
+using ApiSample.Schema;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Serialization;
+
+namespace ApiSample {
+  record ForecastJson(
+    string OperatingArea,
+    DateOnly ForecastStartDate,
+    DateTime UtcRetrievalTimestamp,
+    DateTime UtcForecastTimestamp,
+    bool IsPinned,
+    int Idf,
+    List<ForecastValueJson> LoadForecast
+  ) {}
+
+  record ForecastValueJson(
+    DateOnly Date,
+    int DaysOut,
+    double Forecast
+  ) {} 
+
+  class Program {
+    static string ConnectionString => Environment.GetEnvironmentVariable("CONN_STRING")!;
+    static string ApiKey => System.Environment.GetEnvironmentVariable("MCAST_API_KEY")!;
+    static readonly string MCastDomain = "demo-gas.mea-analytics.tools";
+    static DateOnly StartDate = new DateOnly(2022,01,01);
+    static readonly HttpClient Client = new HttpClient();
+
+    // This lets us use PascalCase for our field names in the records defined above, which is standard for C#
+    // even though the JSON we receive from the API uses camelCase.
+    static readonly JsonSerializerSettings SerializerSettings = 
+      new JsonSerializerSettings() { ContractResolver = new CamelCasePropertyNamesContractResolver() };
+
+    public static async Task Main(string[] args) {
+
+      DotEnv.Load(new DotEnvOptions(probeForEnv:true, probeLevelsToSearch:5));
+
+      Client.DefaultRequestHeaders.Accept.Clear();
+      Client.DefaultRequestHeaders.Add("x-api-key", ApiKey);
+
+      // var recordsToStore = await GetNewForecastRecords();
+
+      // var (nFcstRecordsStored, nFcstValueRecordsStored) = await StoreNewRecords(fcstRecords, fcstValueRecords);
+      // Console.WriteLine($"Wrote {nFcstRecordsStored} forecast records to the database");
+      // Console.WriteLine($"Wrote {nFcstValueRecordsStored} forecast value records to the database");
+    }
+
+    record ForecastData(IEnumerable<LoadForecast> FcstRecords, IEnumerable<LoadForecastValue> FcstValueRecords) { }
+
+    static async Task<ForecastData> GetNewObservationRecords() {
+      throw new NotImplementedException();
+    }
```

Comment:
This would be what we build in the webinar.
Also, oops:
```suggestion
    static async Task<ForecastData> GetNewForecastRecords() {
      throw new NotImplementedException();
    }
```